server/src/main/java/org/elasticsearch/index/snapshots/blobstore/SlicedInputStream.java

```diff
@@ -98,6 +106,29 @@ public final int read(byte[] buffer, int offset, int length) throws IOException
         return read;
     }
 
+    @Override
+    public long skip(long n) throws IOException {
+        long remaining = n;
+        while (remaining > 0) {
+            final InputStream stream = currentStream();
+            if (stream == null) {
+                break;
+            }
+            long skipped = stream.skip(remaining);
+            currentSliceOffset += skipped;
+            if (skipped == 0) {
+                // read one more byte to see if we reached EOF in order to proceed to the next stream.
+                if (stream.read() < 0) {
+                    nextStream();
+                } else {
+                    skipped++;
```

Comment:
Should we also increment `currentSliceOffset` here?

Reply:
Thanks for catching this!